sonar/modules/documents/views.py

```diff
@@ -138,18 +137,15 @@ def project_detail(pid_value, view='global'):
     :param view: Organisation's view.
     :returns: Rendered template.
     """
-    record = ProjectRecord.get_record_by_pid(pid_value)
-
-    if not record:
+    try:
+        record = sonar.resources['projects'].service.record_cls.pid.resolve(
```

Comment:
As most of the time we need the service, we can create a `sonar.service('projects')` or something similar.